lib/core/widgets/custom_chart.dart

```diff
@@ -0,0 +1,282 @@
+import 'package:deus/models/chart_data_point.dart';
+import 'package:deus/models/value_locked_chart_data.dart';
+import 'package:deus/statics/my_colors.dart';
+import 'package:deus/statics/styles.dart';
+import 'package:fl_chart/fl_chart.dart';
+import 'package:flutter/material.dart';
+import 'dart:math';
+
+import 'package:flutter_icons/flutter_icons.dart';
+
+class CustomChart extends StatefulWidget {
+  final double perfInPerc;
+  final double perfInCash;
+  final double lockedInCash;
+  final double lockedInEth;
+  final Future<List<ChartDataPoint>> chartPoints;
```

Comment:
Hier fallen mit zwei Punkte ins Auge:

**a) Mit den aktuellen Feldern ist das Widget nicht allgemein genug.** 

Wir wollen es später auf zwei Screens verwenden:
1. dem WalletScreen
2. dem LockAndStakeOverviewScreen (ich glaub das ist übrigens der, den du aktuell gemacht hast - werden wir also vmtl. noch umbenennen müssen). 
Mit den aktuellen Feldern ist es aber nur für letzteren Screen ausgelegt.

Ich denke die beste Lösung ist, wenn wir einfach im CustomChart nur das reine Chart mit den Time Buttons haben und die Beschriftung und der Rand außenrum dann nochmal in einzelnen Widgets sind, die dann in screens/wallet/widget/ o.ä. liegen. Also dann hast du z.B. ein `Widget LockChart` und das enthält dann sowohl den Hintergrund, eine Instanz von CustomChart (also _diesem_ allgemeinen Widget hier) und eben den Texten mit der Performance und dem gelockten Wert.

**b) Es ist unrealistisch, dass man ein Future mit _allen_ Datensätzen des letzten Jahres standardmäßig erhält.** 
Wenn ich es richtig sehe, übergibst du ein Future mit der Liste von DataPoints, das dann alle Daten enthält und danach filterst du nur noch innerhalb dieser Liste. Das würde aber bedeuten, dass die API später ein Future übergibt, dass IMMER defaultmäßig den größtmöglichen Datensatz zieht - das ist unrealistisch bzw. wäre ziemlich ineffizient. Der häufigste Fall wird sein, dass sich jemand eher für die kleinen Datensätze interessiert, z.B. der letzten Woche. Dementsprechend sollte es eine CallBack Funktion geben, die ausgeführt wird, wenn man auf einen der Zeit-Buttons klickt. Dann wird danach quasi ne neue Anfrage an die API geschickt mit dem Zeitraum, den man haben will und wir kriegen die neuen Daten als Future.


- [ ] CustomChart erhält nur die `Future<List<ChartDataPoint> dataPoints`.
- [ ] CustomChart erhält eine `Future<List<ChartDataPoint> Function(Duration pastDuration) onTimeSelected`, mit der ein neues Future übergeben kann mit den neuen Datensätzen. 
- [ ] Das Ergebnis (das Future) der Funktion sollte dann innerhalb des Widgets wiederverwendet werden, um das Future zu laden und die Daten anzuzeigen.